src/Microsoft.Framework.Runtime.Hosting/Json/JsonContent.cs

```diff
@@ -0,0 +1,199 @@
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace Microsoft.Framework.Runtime.Json
+{
+    /// <summary>
+    /// JsonBuffer represents a piece of loaded json content.
+    /// 
+    /// The JsonBuffer is used in JsonDeserializer only. It is not inteneded to be
+    /// used for data exchange.
+    /// </summary>
+    internal class JsonContent
+    {
+        private List<string> _content = new List<string>();
+
+        /// <summary>
+        /// Create a JsonContent instance from a stream.
+        ///
+        /// Once created the JsonContent instance won't keep the handle to the 
+        /// stream nor dispose or close it.
+        /// </summary>
+        /// <param name="stream">Content source</param>
+        /// <returns>Newly created JsonContent instance</returns>
+        public static JsonContent CreateFromStream(Stream stream)
+        {
+            if (stream == null)
+            {
+                throw new ArgumentNullException(nameof(stream));
+            }
+
+            var content = new List<string>();
+            var reader = new StreamReader(stream);
+
+            string line = reader.ReadLine();
+            while (line != null)
+            {
+                content.Add(line);
+                line = reader.ReadLine();
+            }
+
+            return new JsonContent(content);
+        }
+
+        private JsonContent(List<string> content)
+        {
+            _content = content;
+        }
+
+        public int TotalLines { get { return _content.Count; } }
+
+        /// <summary>
+        /// Current line number in zero-based index.
+        /// </summary>
+        public int CurrentLine { get; private set; } = 0;
+
+        /// <summary>
+        /// Current column number in zero-based index.
+        /// </summary>
+        public int CurrentColumn { get; private set; } = -1;
+
+        public char CurrentChar
+        {
+            get { return _content[CurrentLine][CurrentColumn]; }
+        }
+
+        public bool ValidCursor
+        {
+            get
+            {
+                return CurrentLine < _content.Count &&
+                       CurrentLine >= 0 &&
+                       CurrentColumn < _content[CurrentLine].Length &&
+                       CurrentColumn >= 0;
+            }
+        }
+
+        public bool Started
+        {
+            get { return CurrentLine != 0 || CurrentColumn != -1; }
+        }
+
+        /// <summary>
+        /// Move the cursor to the next non empty char
+        /// </summary>
+        /// <returns>Returns false if the cursor reach the end of the content.</returns>
+        public bool MoveToNextNonEmptyChar()
+        {
+            while (TotalLines > CurrentLine)
+            {
+                while (_content[CurrentLine].Length > CurrentColumn + 1)
+                {
+                    char c = _content[CurrentLine][++CurrentColumn];
+                    if (!char.IsWhiteSpace(c))
```

Comment:
Be careful with these `char` helper methods - they rarely do what you want. Please check the JSON / JavaScript specs to see exactly what *they* define as whitespace and use that. It's *possible* that they define it in the same way that Unicode does (which is what `char.IsWhiteSpace` uses), but that's generally unlikely.